app/src/main/java/nl/tudelft/bbw/API.java

```diff
@@ -32,75 +40,105 @@ private API() {
      * can only be done once per user
      * TODO: Remove this function
      *
-     * @param owner   the User
+     * @param Name    of the API user
+     * @param Iban    of the API user
      * @param context The state of the program
      * @throws HashException               When creating a block results in an error
      * @throws BlockAlreadyExistsException when adding a block results in an error
      */
-    public static void initializeAPI(User owner, Context context) throws HashException, BlockAlreadyExistsException {
+    public static User initializeAPI(String Name, String Iban, Context context) throws HashException, BlockAlreadyExistsException {
+
+        User owner;
+        EdDSAPrivateKey privateKey = ED25519.generatePrivateKey();
+        owner = new User(Name, Iban, ED25519.getPublicKey(privateKey));
+        owner.setPrivateKey(privateKey);
         blockController = new BlockController(owner, context);
         blockVerificationController = new BlockVerificationController(context);
         blockController.createGenesis(owner);
+        return owner;
     }
 
     /**
-     * Method to add a contact to your chain
+     * Getter method to get the API user's contact list
      *
-     * @param contact the contact
-     * @param signature byte array containing the signature
-     * @param message byte array containing the message
-     * @throws HashException               When creating a block results in an error
-     * @throws BlockAlreadyExistsException when adding a block results in an error
+     * @return List of blocks forming this user contact list.
      */
-    public static void addContactToChain(User contact, byte[] signature, byte[] message)
-            throws HashException, BlockAlreadyExistsException {
-        blockController.addBlockToChain(contact, signature, message);
+    public static List<Block> getMyContacts() {
+        return blockController.getBlocks(blockController.getOwnUser());
     }
 
     /**
-     * Method to revoke a contact to your chain
+     * Getter method to get the API user's name
      *
-     * @param contact the contact
-     * @throws HashException               When creating a block results in an error
-     * @throws BlockAlreadyExistsException when adding a block results in an error
+     * @return name of the API user
      */
-    public static void revokeContactFromChain(User contact)
-            throws HashException, BlockAlreadyExistsException {
-        blockController.revokeBlockFromChain(contact);
+    public static String getMyName() {
+        return blockController.getOwnUser().getName();
```

Comment:
Why do you have so much getters; whilst you can just create one getter for a user, and get attributes from that user.